File: framework/languages/java/samples/java/TicTacToe.SessionGateway/Server/Api/src/main/java/systems/zlink/samples/tictactoe/sessiongateway/server/api/ApiServerHostFactory.java
package systems.zlink.samples.tictactoe.sessiongateway.server.api;

import org.springframework.boot.WebApplicationType;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.builder.SpringApplicationBuilder;
import org.springframework.context.annotation.Bean;
import systems.zlink.framework.spring.ZLinkFrameworkOptionsCustomizer;
import systems.zlink.samples.tictactoe.sessiongateway.shared.configuration.SampleTopology;

@SpringBootApplication(
    proxyBeanMethods = false,
    scanBasePackageClasses = ApiServer.class)
public final class ApiServerHostFactory {
    private ApiServerHostFactory() {
    }

    public static AutoCloseable start(String... args) {
        SpringApplicationBuilder builder = new SpringApplicationBuilder(ApiServerHostFactory.class)
            .web(WebApplicationType.NONE);
        builder.application().setKeepAlive(true);
        return builder.run(args)::close;
    }

    @Bean
    ZLinkFrameworkOptionsCustomizer apiOptions() {
        return options -> {
            options.addRegistryEndpoint(SampleTopology.RegistryRouterEndpoint);
            ApiServer.configure(options);
        };
    }
}

File: framework/languages/java/samples/kotlin/TicTacToe.SessionGateway/Server/Api/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/sessiongateway/server/api/ApiServerHostFactory.kt
package systems.zlink.samples.kotlin.tictactoe.sessiongateway.server.api

import org.springframework.boot.WebApplicationType
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.builder.SpringApplicationBuilder
import org.springframework.context.annotation.Bean
import systems.zlink.framework.spring.ZLinkFrameworkOptionsCustomizer
import systems.zlink.samples.kotlin.tictactoe.sessiongateway.shared.configuration.SampleTopology

@SpringBootApplication(
    proxyBeanMethods = false,
    scanBasePackageClasses = [ApiServer::class],
)
class ApiServerHostFactory {
    @Bean
    fun apiOptions(): ZLinkFrameworkOptionsCustomizer =
        ZLinkFrameworkOptionsCustomizer { options ->
            options.useDiscovery { discovery ->
                discovery.addRegistryEndpoint(SampleTopology.RegistryRouterEndpoint)
            }
            ApiServer.configure(options)
        }

    companion object {
        fun start(args: Array<String> = emptyArray()): AutoCloseable {
            val builder = SpringApplicationBuilder(ApiServerHostFactory::class.java)
                .web(WebApplicationType.NONE)
            builder.application().setKeepAlive(true)
            val context = builder.run(*args)
            return AutoCloseable { context.close() }
        }
    }
}
